### DISCUSSION TITLE
kalendar.pl

### DISCUSSION THREAD

**Discussion by mbab** [DivinumOfficium/divinum-officium#2569] [Developers corner]
Is there any purpose to have separate `kalendar.pl` for officium and missa?
Output look almost same but maybe they should differ?
I see not much work to unificate them and also probably also abandon both `Ckalendar.pl`.

**Comment by ajmalton** [comment]
I think that the Kalendar modules and result should be _unified_ where there are presentation or content differences, but the logic should be _one_.

After all, the Roman Kalendar is one, be it for Mass or for Office.

I think the Compare logic should _not_ be abandoned, because it's essential to Kiss's original concept: differences within essential unity.  

However, the Compare logic, like the multilingual logic, is currently brittle and difficult, because the content and the presentation are computed simultaneously.  Thus all the switching of globals and columns between 1 and 2, etc etc.  It would be far better, more correct and maintainable design, to compute the content for two columns separately (be they two versions or two Kalendars or two languages) and then present them together.  This is going to be possible, but the modernization changes have to be done first.

(I have 2000 files of regression test and my current revision, with use warnings, use strict, and no requires, passes them all -- in other words, no output changes at all yet.  But this revision should pass 20,000 tests before it moves to production!)

**Comment by mbab** [comment]
I made attempt to unificate kalendar for officium, missa and 'compare'.
https://github.com/mbab/divinum-officium/tree/hOneKalendar
please check if it can be merged now or rather wait till 'modularization' is finished.
(missa/kalendar.pl and both Ckalendar.pl are obsolete but probably should be replaced
with redirections)

**Reply by ajmalton** [reply]
You have done all this, including the redirections, and it works, and now it is seems that `missa/*kalendar.pl` and `horas/Ckalendar.pl` -- three files -- can be removed entirely.
